### PR TITLE
Improve development manfests and docs

### PR DESCRIPTION
**Pull request by Tomcli**
**Which issue is resolved by this Pull Request:** 
Resolves #

**Description of your changes:**
Since the original KFP dev manifests and docs are not very friendly for new comers, I did some improvement based on my experience and convert it in a way that can be installed with `kubectl -k` alone. 

Also, I changed few minor things in the dev manifests:
- removed Argo since we don't need it until the main KFP repo can upgrade their Argo version.
- The dev images are pointing to latest which is built on the latest commit.

**Environment tested:**

* Python Version (use `python --version`): 3.7
* Tekton Version (use `tkn version`): 0.15
* Kubernetes Version (use `kubectl version`): 1.16
* OS (e.g. from `/etc/os-release`):

**Checklist:**
- [ ] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention). 

   PR titles examples:
    * `fix(frontend): fixes empty page. Fixes #1234`
       Use `fix` to indicate that this PR fixes a bug.
    * `feat(backend): configurable service account. Fixes #1234, fixes #1235`
       Use `feat` to indicate that this PR adds a new feature. 
    * `chore: set up changelog generation tools`
       Use `chore` to indicate that this PR makes some changes that users don't need to know.
    * `test: fix CI failure. Part of #1234`
        Use `part of` to indicate that a PR is working on an issue, but shouldn't close the issue when merged.

- [ ] Do you want this pull request (PR) cherry-picked into the current release branch?
    
    If yes, use one of the following options:
  
    * **(Recommended.)** Ask the PR approver to add the `cherrypick-approved` label to this PR. The release manager adds this PR to the release branch in a batch update.
    *  After this PR is merged, create a cherry-pick PR to add these changes to the release branch. (For more information about creating a cherry-pick PR, see the [Kubeflow Pipelines release guide](https://github.com/kubeflow/pipelines/blob/master/RELEASE.md#option--git-cherry-pick).)

### CODE REVIEW

**Comment by Tomcli** [comment]
@ckadner can you review this? thanks

File: developer_guide.md [hunk 1]
```diff
@@ -3,40 +3,43 @@
...
+    Once all the pods except Cache deployer are running, run the commands below to expose your KFP UI to a public endpoints.
+    ```shell
+    kubectl patch svc ml-pipeline-ui -n kubeflow -p '{"spec": {"type": "LoadBalancer"}}'
+    kubectl -n kubeflow get service ml-pipeline-ui -o jsonpath='{.status.loadBalancer.ingress[0].ip}'
+    ```
 
-    Once you have Kubeflow Pipelines running with Tekton, then install the [KFP-Tekton SDK](/sdk/README.md) and start building your
-    own pipelines.
+    Once you have Kubeflow Pipelines running with Tekton, then install the [KFP-Tekton SDK](/sdk/README.md) and learn about the
+    [KFP Tekton User Guide](/samples/kfp-user-guide) to start building your own pipelines.
```
Comment (ckadner) [inline developer_guide.md:42]: I would slightly rephrase this to:
> Now that you have deployed Kubeflow Pipelines with Tekton, install the [KFP-Tekton SDK](/sdk/README.md) and follow the [KFP Tekton User Guide](/samples/kfp-user-guide) to start building your own pipelines.

**Review by ckadner — COMMENTED** [review]
Looks good overall. Just a few wording change suggestions :-)

File: developer_guide.md [hunk 2]
```diff
@@ -3,40 +3,43 @@
...
 
-2. Once the old KFP deployment is removed, run the below command to deploy this modified Kubeflow Pipeline version with Tekton backend.
+2. Once the previous KFP deployment is removed, run the below command to deploy this modified Kubeflow Pipeline version with Tekton backend.
     ```shell
+    kubectl create ns kubeflow # Skip this if kubeflow namespace aleady exists
     kubectl apply -k manifests/kustomize/env/platform-agnostic
     ```
 
-    Check the new KFP deployment, it should take about 5 to 10 minutes.
+    Check the new KFP deployment, it should take about 5 to 10 minutes. Please be aware that [cache-deployer-deployment](https://www.kubeflow.org/docs/pipelines/caching/) won't be running unless it's deployed on top of the whole [Kubeflow stack](https://www.kubeflow.org/docs/started/getting-started/) with [Cert Manager](https://cert-manager.io/docs/installation/kubernetes/).
```
[the remark this answers is not in the document]
Comment (Tomcli) [reply developer_guide.md]: done

**Comment by Tomcli** [comment]
Thanks @ckadner for your review.

**Review by ckadner — APPROVED** [review]
/lgtm
